### ISSUE TITLE
Release the source code

### ISSUE THREAD

**Issue by hubertbanas** [roomsense/firmware#14]
Please release the source code.

**Comment by talmuth** [comment]
Is there a reason why this can't be done? I'd like to tinker with all the devices I've invested into

**Comment by russellmilliner** [comment]
They've gone completely quiet.  They delivered the product and it seems like they may be done.  It was so promising.  I'd like to be proven wrong.

**Comment by talmuth** [comment]
I've spent some time on replacing stock firmware with ESPHome, was able to initialize almost all the sensors except CO and PMS on the ClimateSense. 
- It Iooks like CO sensor is attached to the pin (GPIO20) that can't be used for ADC if WiFi is enabled, so I'm not sure if it even measures anything on stock firmware
- Installed PM sensor, doesn't have support in ESPHome out of the box, so I started to work on the new integration for it as well, but as I'm not yet familiar enough with i2c, I couldn't quite figure out how to talk to sensor.

**Comment by russellmilliner** [comment]
Maybe we should take these further research/discussion to a different forum like Discord?

**Comment by middelink** [comment]
@talmuth I'm not sure that is the case on an ESP-32 S3. At least the warning disappeared if you compare the [S3 gpio summary](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/peripherals/gpio.html) with the [esp32 gpio summary](https://docs.espressif.com/projects/esp-idf/en/latest/esp32s3/api-reference/peripherals/gpio.html).

or in fact the hardware limitations of the ADC. https://docs.espressif.com/projects/esp-idf/en/latest/esp32s3/api-reference/peripherals/adc_continuous.html#hardware-limitations

Edit: I think the PM might be the MPM10 from memfs https://robu.in/wp-content/uploads/2024/01/DataSheet.pdf. (Guestimated from my i2c dumps where it reads 0x4d, register 0x20-0x25 and some googling)

**Comment by talmuth** [comment]
@middelink yeah, that seems to be a limitation of esphome, still need to file a issue for them. But if you disable the check, it compiles and just "works". Still need to figure out how to do a mapping of voltage that it receives to ppm, given actual signal from the sensor gets amplified by 4 op-ams series before it reaches esp32

**Comment by middelink** [comment]
We are talking about the CO sensor right? Yeah, the [datasheet](https://www.figaro.co.jp/en/product/docs/tgs5141-p00_product%20infomation%28en%29_rev05.pdf) is exceptionally vague. well, 1.2~3.2nA/ppm. Not helpful.

OT, were you able to get a something running on it? Using esphome 2024.12.2, it leaves me with a device where I have like >90% ping loss on the wifi. Putting the original firmware back makes it ping without loss. So soomething is going wrong either in the setup of the sensors, or the newer idf is annoying on S3's.

**Comment by talmuth** [comment]
@middelink I'm not the OT, but I suppose you were asking me. Yes, the one that I was playing with is currently running 2024.12.2, I don't see any loses

**Comment by roomsense** [comment]
[Here](https://github.com/roomsense/roomsense-iq/tree/main/firmware) you can find the open source code of RoomSense IQ.
